sdv/lite/single_table.py

```diff
@@ -0,0 +1,250 @@
+"""Base class for single table model presets."""
+
+import logging
+import sys
+
+import cloudpickle
+import numpy as np
+import rdt.transformers
+
+from sdv.single_table import GaussianCopulaSynthesizer
+from sdv.utils import get_package_versions, throw_version_mismatch_warning
+
+LOGGER = logging.getLogger(__name__)
+
+FAST_ML_PRESET = 'FAST_ML'
+PRESETS = {
+    FAST_ML_PRESET: 'Use this preset to minimize the time needed to create a synthetic data model.'
+}
+
+
+class SingleTablePreset():
+    """Class for all single table synthesizer presets.
+
+    Args:
+        metadata (sdv.metadata.SingleTableMetadata):
+            ``SingleTableMetadata`` instance.
+        name (str):
+            The preset to use.
+    """
+
+    _synthesizer = None
+    _null_percentages = None
+    _null_column = False
+    _default_synthesizer = GaussianCopulaSynthesizer
+
+    def __init__(self, metadata, name):
+        if name not in PRESETS:
+            raise ValueError(f'`name` must be one of {PRESETS}.')
+
+        self.name = name
+
+        if name == FAST_ML_PRESET:
+            self._synthesizer = GaussianCopulaSynthesizer(
+                metadata=metadata,
+                default_distribution='norm',
+                enforce_rounding=False
+            )
+            self._synthesizer._data_processor._update_transformers_by_sdtypes(
+                'categorical',
+                rdt.transformers.FrequencyEncoder(add_noise=True)
+            )
+
+            # Decide if transformers should model the null column or not.
+            self._null_column = len(metadata.to_dict().get('constraints', [])) > 0
+
+            # Update transformers to model missing values if necessary
+            transformer_model_missing_values = bool(self._null_column)
+
+            sdtype_transformers = {
+                'numerical': rdt.transformers.FloatFormatter(
+                    missing_value_replacement='mean',
+                    model_missing_values=transformer_model_missing_values,
+                    enforce_min_max_values=True,
+                ),
+                'categorical': rdt.transformers.FrequencyEncoder(add_noise=True),
+                'boolean': rdt.transformers.BinaryEncoder(
+                    missing_value_replacement=-1 if self._null_column else 'mode',
+                    model_missing_values=transformer_model_missing_values,
+                ),
+                'datetime': rdt.transformers.UnixTimestampEncoder(
+                    missing_value_replacement='mean' if self._null_column else 'mode',
+                    model_missing_values=transformer_model_missing_values,
+                ),
+            }
+            self._synthesizer._data_processor._transformers_by_sdtype.update(sdtype_transformers)
```

Comment:
I think that we should move this logic outside the `__init__`, otherwise if we add more presents in the future this will get bigger and bigger.
I would recommend a function that accepts the `name` and then proceeds with the preset.